data_structures/linked_list/stack_using_linked_lists.c

```diff
@@ -36,40 +36,51 @@ int main()
     }
 }
 
-void push(struct node *p)
+/**
+ * push function will add a new node at the head of the list, time complexity O(1).
+ * @param p pointer to the top of the stack.
+ * @returns void.
+ */
+void push(struct node *p) 
 {
     int item;
     struct node *temp;
-    temp = (struct node *)malloc(sizeof(struct node));
+    temp = (struct node *)malloc(sizeof(struct node)); 
```

Comment:
Please remove this empty whitespace.
Same in other parts of the code.
```suggestion
    temp = (struct node *)malloc(sizeof(struct node));
```